AlphaWallet/KeyManagement/EtherKeystore.swift

```diff
@@ -0,0 +1,647 @@
+        return .success(())
+    }
+
+    private func deletePrivateKeysFromSecureEnclave(forAccount account: EthereumAccount) {
+        let secureEnclave = SecureEnclave()
+        secureEnclave.deletePrivateKeys(withName: encryptionKeyForPrivateKeyLabel(fromAccount: account, withUserPresence: true))
+        secureEnclave.deletePrivateKeys(withName: encryptionKeyForPrivateKeyLabel(fromAccount: account, withUserPresence: false))
+        secureEnclave.deletePrivateKeys(withName: encryptionKeyForSeedLabel(fromAccount: account, withUserPresence: true))
+        secureEnclave.deletePrivateKeys(withName: encryptionKeyForSeedLabel(fromAccount: account, withUserPresence: false))
+    }
+
+    private func deleteKeysAndSeedCipherTextFromKeychain(forAccount account: EthereumAccount) {
+        keychain.delete("\(Keys.ethereumRawPrivateKeyUserPresenceNotRequiredPrefix)\(account.address.eip55String)")
+        keychain.delete("\(Keys.ethereumRawPrivateKeyUserPresenceRequiredPrefix)\(account.address.eip55String)")
+        keychain.delete("\(Keys.ethereumSeedUserPresenceNotRequiredPrefix)\(account.address.eip55String)")
+        keychain.delete("\(Keys.ethereumSeedUserPresenceRequiredPrefix)\(account.address.eip55String)")
+    }
+
+    private func removeAccountFromBookkeeping(_ account: EthereumAccount) {
+        ethereumAddressesWithPrivateKeys = ethereumAddressesWithPrivateKeys.filter { $0 != account.address.eip55String }
+        ethereumAddressesWithSeed = ethereumAddressesWithSeed.filter { $0 != account.address.eip55String }
+        ethereumAddressesProtectedByUserPresence = ethereumAddressesProtectedByUserPresence.filter { $0 != account.address.eip55String }
+        watchAddresses = watchAddresses.filter { $0 != account.address.eip55String }
+    }
+
+    func delete(wallet: Wallet, completion: @escaping (Result<Void, KeystoreError>) -> Void) {
+        DispatchQueue.global(qos: .userInitiated).async { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+            let result = strongSelf.delete(wallet: wallet)
+            DispatchQueue.main.async {
+                completion(result)
+            }
+        }
+    }
+
+    func isHdWallet(account: EthereumAccount) -> Bool {
+        return ethereumAddressesWithSeed.contains(account.address.eip55String)
+    }
+
+    func isHdWallet(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithSeed.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isKeystore(wallet: Wallet) -> Bool {
+        switch wallet.type {
+        case .real(let account):
+            return ethereumAddressesWithPrivateKeys.contains(account.address.eip55String)
+        case .watch:
+            return false
+        }
+    }
+
+    func isWatched(wallet: Wallet) -> Bool {
```

Comment:
What exactly does isWatched mean?

Reply:
User is observing an address for which he does not have a key.